testcases/PowerNVDump.py

```diff
@@ -0,0 +1,418 @@
+                # if both fadump and kdump is disabled, system should do normal
+                # IPL after kernel crash(oops)
+            except KernelPanic:
+                print "==================Normal Boot==========================="
+                # if both fadump and kdump is disabled, system should do
+                # normal IPL after kernel crash(oops)
+                boot_type = BootType.NORMAL
+            except KernelKdump:
+                print "================Kdump kernel boot======================="
+                # if only kdump is enabled, kdump kernel should boot
+                # after kernel crash
+                boot_type = BootType.KDUMPKERNEL
+            except KernelCrashUnknown:
+                self.cv_SYSTEM.set_state(OpSystemState.UNKNOWN_BAD)
+                done = True
+                boot_type = BootType.NORMAL
+            except pexpect.TIMEOUT:
+                done = True
+                boot_type = BootType.INVALID
+                self.cv_SYSTEM.set_state(OpSystemState.UNKNOWN_BAD)
+            except PlatformError:
+                done = True
+                boot_type = BootType.NORMAL
+            if rc == 0:
+                self.cv_SYSTEM.set_state(OpSystemState.IPLing)
+                done = True
+            if rc == 1 or rc == 2:
+                print "Kdump finished collecting core file, waiting for regular IPL to complete"
+
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        #self.cv_HOST.ssh.state = SSHConnectionState.DISCONNECTED
+        print "System booted fine to host OS..."
+        return boot_type
+
+
+class OPALCrash_MPIPL(PowerNVDump):
+
+    def runTest(self):
+        if not self.pdbg or not os.path.exists(self.pdbg):
+            self.fail("pdbg file %s doesn't exist" % self.pdbg)
+        if not self.is_fadump_param_enabled():
+            raise self.skipTest(
+                "fadump=on not added in kernel param, please add and re-try")
+        self.setup_test()
+        cmd = "rm /tmp/pdbg; rm /tmp/deadbeef"
+        try:
+            self.cv_BMC.run_command(cmd)
+        except CommandFailed:
+            pass
+        # copy the pdbg file to BMC
+        self.cv_BMC.image_transfer(self.pdbg)
+        deadbeef = os.path.join(self.basedir, "test_binaries", "deadbeef")
+        self.cv_BMC.image_transfer(deadbeef)
+        if "OpenBMC" in self.bmc_type:
+            cmd = "/tmp/pdbg -a putmem 0x300000f8 < /tmp/deadbeef"
+        elif "SMC" in self.bmc_type:
+            cmd = "/tmp/rsync_file/pdbg -a putmem 0x300000f8 < /tmp/rsync_file/deadbeef"
+        self.cv_BMC.run_command(cmd)
+        done = False
+        boot_type = BootType.NORMAL
+        rc = -1
+        while not done:
+            try:
+                # MPIPL completion + system reboot would take time, keeping it
+                #600 seconds.
+                rc = self.c.pty.expect(['\n', 'ISTEP 10.1'], timeout=600)
```

Comment:
Add comment explaining use of "ISTEP 10.1 " here . Otherwise its hard to understand.